File: articles/biochemistry-molecular-basis-life/fortran/biochemistry_kernel.f90
program biochemistry_kernel
  implicit none

  real(8) :: velocity, occupancy, hill_occupancy, delta_g
  real(8), parameter :: R = 8.314462618d0

  velocity = 120.0d0 * 5.0d0 / (3.5d0 + 5.0d0)
  occupancy = 2.0d0 / (2.0d0 + 2.0d0)
  hill_occupancy = (2.0d0 ** 2.0d0) / ((2.0d0 ** 2.0d0) + (2.0d0 ** 2.0d0))
  delta_g = -(R * 298.15d0 * log(1000.0d0)) / 1000.0d0

  print *, "velocity=", velocity
  print *, "occupancy=", occupancy
  print *, "hill_occupancy=", hill_occupancy
  print *, "delta_g_kj_mol=", delta_g

end program biochemistry_kernel

File: articles/biochemistry-molecular-basis-life/julia/biochemistry_kernel.jl
# Biochemistry numerical kernel in Julia.

function michaelis_menten(substrate, vmax, km)
    return vmax * substrate / (km + substrate)
end

function occupancy(ligand, kd)
    return ligand / (kd + ligand)
end

function hill_occupancy(ligand, kd, n)
    return ligand^n / (kd^n + ligand^n)
end

function delta_g_standard(K, T)
    R = 8.314462618
    return -(R * T * log(K)) / 1000.0
end

println("velocity=", round(michaelis_menten(5.0, 120.0, 3.5), digits=6))
println("occupancy=", round(occupancy(2.0, 2.0), digits=6))
println("hill_occupancy=", round(hill_occupancy(2.0, 2.0, 2.0), digits=6))
println("delta_g_kj_mol=", round(delta_g_standard(1000.0, 298.15), digits=6))
